### PR TITLE
Get enclaves refactoring working with minimal changes to lingua-franca

### PR DESCRIPTION
**Pull request by edwardalee**
The enclaves2 branch is coupled with substantial changes to lingua-franca (AST transformations to create hierarchy). This branch is trying to evaluate what minimal changes to lingua-franca are required to get things to work, preserving the changes in the reactor-c enclaves2 branch.

### CODE REVIEW

File: core/federated/RTI/rti_common.c [hunk 1]
```diff
@@ -10,6 +10,8 @@
 #if defined STANDALONE_RTI || defined LF_ENCLAVES
 #include "rti_common.h"
 
+extern lf_mutex_t rti_mutex;
```
Comment (erlingrj) [inline core/federated/RTI/rti_common.c:13]: I dont think this is needed. 

In `initailize_rti`, both in `rti_remote.c` and `rti_local.c` the `mutex` field of the `rti_common` is set. So we should be able to just use this field?

File: core/federated/RTI/rti_common.c [hunk 2]
```diff
@@ -115,6 +117,7 @@ tag_advance_grant_t tag_advance_grant_if_safe(scheduling_node_t* e) {
     // To handle cycles, need to create a boolean array to keep
     // track of which upstream enclave have been visited.
     bool *visited = (bool *)calloc(rti_common->number_of_scheduling_nodes, sizeof(bool)); // Initializes to 0.
+    visited[e->id] = true;
```
Comment (erlingrj) [inline core/federated/RTI/rti_common.c:120]: This and the corresponding change in `rti_remote.c` seems to cause a test failure in `FeedbackDelay.lf`. Or, at least, that tests passes on the original `enclaves2` branch. I dont see this being done on `main` either?

**Comment by erlingrj** [comment]
I have a fix for the Arduino failure on the `enclaves2` LF branch. It is due to a very inflexible build system for Arduino...

**Comment by edwardalee** [comment]
Closing this because enclaves2 has been merged so it makes no sense to merge this into enclaves2.

**Pull request closed without merging**